iota-client/src/node/mqtt.rs

```diff
@@ -0,0 +1,207 @@
+    Self::new(value)
+  }
+}
+
+impl Topic {
+  /// Creates a new topic and checks if it's valid.
+  pub fn new<S: Into<String>>(name: S) -> Result<Self> {
+    let valid_topics = lazy_static!(
+      ["milestones/latest", "milestones/solid", "messages", "messages/referenced"].to_vec() => Vec<&str>
+    );
+    let regexes = lazy_static!(
+      [
+        Regex::new(r"messages/([A-Fa-f0-9]{64})/metadata").unwrap(),
+        Regex::new(r"outputs/([A-Fa-f0-9]{64})(\d{4})").unwrap(),
+        Regex::new("addresses/([A-Fa-f0-9]{64})/outputs").unwrap(),
+        Regex::new(r"messages/indexation/(\.)").unwrap()
+      ].to_vec() => Vec<Regex>
+    );
+    let name = name.into();
+    if valid_topics.iter().any(|valid| valid == &name)
+      || regexes.iter().any(|re| re.is_match(&name))
+    {
+      let topic = Self(name);
+      Ok(topic)
+    } else {
+      Err(crate::Error::InvalidMqttTopic(name))
+    }
+  }
+}
+
+pub(crate) fn get_mqtt_client(client: &mut Client) -> Result<&MqttClient> {
+  match client.mqtt_client {
+    Some(ref c) => Ok(c),
+    None => {
+      for node in client.pool.read().unwrap().iter() {
+        // node.set_path("mqtt");
+        let uri = &format!(
+          "{}://{}:{}/mqtt",
+          if node.scheme() == "https" {
+            "wss"
+          } else {
+            "ws"
+          },
+          node.host_str().unwrap(),
+          node.port_or_known_default().unwrap()
+        );
+        let mqtt_options = CreateOptionsBuilder::new()
+          .server_uri(uri)
+          .client_id(format!("iota.rs-mqtt-client-{}", uri))
+          .finalize();
+        let mut mqtt_client = MqttClient::new(mqtt_options)?;
+
+        let conn_opts = ConnectOptionsBuilder::new()
+          .keep_alive_interval(Duration::from_secs(20))
+          .mqtt_version(MQTT_VERSION_3_1_1)
+          .clean_session(true)
+          .finalize();
+
+        if let Ok(_) = mqtt_client.connect(conn_opts) {
+          poll_mqtt(client.mqtt_topic_handlers.clone(), &mut mqtt_client);
+          client.mqtt_client = Some(mqtt_client);
+          break;
+        }
+      }
+      client
+        .mqtt_client
+        .as_ref()
+        .ok_or_else(|| crate::Error::MqttConnectionNotFound)
+    }
+  }
+}
+
+fn poll_mqtt(mqtt_topic_handlers: Arc<Mutex<TopicHandlerMap>>, client: &mut MqttClient) {
+  let receiver = client.start_consuming();
+  std::thread::spawn(move || {
+    while let Ok(message) = receiver.recv() {
+      if let Some(message) = message {
+        let topic = message.topic().to_string();
+        let mqtt_topic_handlers_guard = mqtt_topic_handlers.lock().unwrap();
+        if let Some(handlers) = mqtt_topic_handlers_guard.get(&Topic(topic.clone())) {
+          let event = TopicEvent {
+            topic,
+            payload: message.payload_str().to_string(),
+          };
+          for handler in handlers {
+            handler(&event)
+          }
+        }
+      }
```

Comment:
Can you help check the while let loop will break when the all the topics are unsubscribed? (Like the mpsc `Receiver` will be dropped.) And it might be good to handle the corresponding Sender is disconnected.